### PR TITLE
ci: setup

### LINKED ISSUE #8: Add and end-to-end test runned by a CI to make sure we can always run an experiment
## Target
Make sure we can always run an experiment without error

## Tasks

- [x] add CI
   - [x] add a .g`ithub/workflows/devops-ci.yml` to configure the CI job
   - [x] add a Makefile with instructions to
      - [x] build the venv
      - [x] isntall requirements
      - [x] activate venv and run unit tests
- [x] add a unit test for rice irrigation problem

### PR DESCRIPTION
**Pull request by akouete-kpakpo**
Closes #8

### CODE REVIEW

**Review by akouete-kpakpo — COMMENTED** [review]
ok @chrichri17 you can review a posteriori but I'm going to merge because I want to test the CI and I can not do if these changes are not merged

File: devops/makefiles/log.mk [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Log
+# ---
+
+echo-cyan:  ## echo with cyan font https://gist.github.com/iamnewton/8754917
+	@echo -e "\e[36m${msg}\e[39m"
```
Comment (chrichri17) [inline devops/makefiles/log.mk:5]: Nice 💯 
Not sure why but it is not working in my terminal (`zsh 5.8 (x86_64-apple-darwin20.0)`).

Running `echo -e "\e[36mHello, World\e[39m"` works as expected but when I run `make test-all`, it just display `\e[36m2.1. Unit tests\e[39m` without coloring the text ^^

File: nevergrad/functions/irrigation/irrigation.py [hunk 2]
```diff
@@ -10,77 +10,69 @@
...
 class Irrigation(ArrayExperimentFunction):
     variant_choice = {}
-    def __init__(self, symmetry: int) -> None:
-        data_dir = Path(__file__).with_name("data")
+
+    def __init__(self, symmetry: int, n_iterations: int = 1000) -> None:
         urllib.request.urlretrieve(
-           "https://raw.githubusercontent.com/ajwdewit/pcse_notebooks/master/data/soil/ec3.soil",
-           Path(IRRIGATION_DATA_DIR, "soil/ec3.soil"),
+            "https://raw.githubusercontent.com/ajwdewit/pcse_notebooks/master/data/soil/ec3.soil",
```
Comment (chrichri17) [inline nevergrad/functions/irrigation/irrigation.py:42]: should we make this retrieve only when the data is not already downloaded ? or only if the user specifies it ?

File: nevergrad/functions/irrigation/known_geoloc.json [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    },
+    "Yaounde": {
+        "latitude": 3.8689867,
+        "longitude": 11.5213344
+    },
+    "Kiev": {
+        "latitude": 50.4500336,
+        "longitude": 30.5241361
+    }
+}
```
Comment (chrichri17) [inline nevergrad/functions/irrigation/known_geoloc.json:46]: nit: end of line

File: nevergrad/functions/irrigation/known_geoloc.json [hunk 4]
```diff
@@ -0,0 +1,46 @@
+{
```
Comment (chrichri17) [inline nevergrad/functions/irrigation/known_geoloc.json:1]: this is actually better than hard coding it in code ✨

**Review by chrichri17 — COMMENTED** [review]
lgtm !

File: devops/makefiles/log.mk (hunk 1 above)
Comment (kevink-oe) [reply devops/makefiles/log.mk:5]: aha, thanks for the feedback @chrichri17 . I'm using a bash terminal. Color works on my side and on CI as well. Maybe you should use a bash terminal?

But if you have a fix that works for every terminal, feel free to open an issue and a MR

File: nevergrad/functions/irrigation/known_geoloc.json (hunk 3 above)
Comment (kevink-oe) [reply nevergrad/functions/irrigation/known_geoloc.json:46]: actually I parsed with black. do you suggest to add a blank line at the end of the json file?

File: nevergrad/functions/irrigation/irrigation.py (hunk 2 above)
Comment (kevink-oe) [reply nevergrad/functions/irrigation/irrigation.py:42]: I would say the first option. But let's keep it simple for the moment. As the download time is quite quick we can just download at every moment the program is runned.

File: nevergrad/functions/irrigation/known_geoloc.json (hunk 4 above)
Comment (kevink-oe) [reply nevergrad/functions/irrigation/known_geoloc.json:1]: aha thanks! I think this code is quite unreadable and I will do some additional refactoring to make it more understandable.

**Comment by kevink-oe** [comment]
perfect @chrichri17 . Thanks for the review. Much appreciated.

File: nevergrad/functions/irrigation/known_geoloc.json (hunk 3 above)
Comment (chrichri17) [reply nevergrad/functions/irrigation/known_geoloc.json:46]: Yeah ! I format json files with prettier and it usually adds a new line at the end.

**Pull request merged**